### DISCUSSION TITLE
How to customize onConnect for automatic reconnection?

### DISCUSSION THREAD

**Discussion by loremus** [kaandesu/vue-paho-mqtt#94] [Q&A]
I am initially connecting to the broker using the following code:

```
$mqtt.connect({
  onConnect: () => {
    // Custom logic
  },
  onFailure: () => {
    // Custom logic
  },
  onConnectionLost: (error: any) => {
    // Custom logic
  },
  onMessageArrived: (message: { payloadString: string; destinationName: string }) => {
    // Custom logic
  }
});
```

Each field in the connection configuration is highly customized. However, during an automatic reconnect, it seems like a default `mqtt.connect()` is executed without any custom callbacks.

Is it possible to either:

- Customize the auto-reconnect `.connect()` method, or
- Force the automatic reconnect to use the initial connection configuration provided?

Any guidance or insights on how to achieve this would be appreciated. Thank you!

**Comment by kaandesu** [comment]
Hey there, thanks for reaching out. Interesting point, there might be a workaround right now, but ill check what I can do on the library side first, I am not available in the next few days but will return to this once I am.

Meanwhile, if you have any time: CC: @EgeOnder

**Comment by newJuly** [comment]
Hey, boss, did you solve the problem
